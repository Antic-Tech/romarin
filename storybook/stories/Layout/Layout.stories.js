import { storiesOf } from '@storybook/react-native';
import React from 'react';
import { Checkbox, Column, Row, Text, Touchable } from '../../components';
import Divider from '../Divider/Divider';

storiesOf('Layout', module).add('List item', () => (
	<React.Fragment>
		<Touchable type="highlight">
			<Row>
				<Column padding="Small" center>
					<Checkbox checked />
				</Column>
				<Column grow paddingTop="Small">
					<Text>Title</Text>

					<Text muted size="Small" paddingBottom="Small">
						Subtitle
					</Text>
					<Divider />
				</Column>
			</Row>
		</Touchable>
		<Touchable>
			<Row>
				<Column padding="Small" center>
					<Checkbox checked />
				</Column>
				<Column grow paddingTop="Small">
					<Text>Title</Text>

					<Text muted size="Small" paddingBottom="Small">
						Subtitle
					</Text>
					<Divider />
				</Column>
			</Row>
		</Touchable>
	</React.Fragment>
));
