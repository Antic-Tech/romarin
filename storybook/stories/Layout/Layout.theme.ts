export default (core) => ({
	textColor: core.colorBlack,
	textColorMuted: core.colorLightGray,
	textColorTouchable: core.colorPrimary,
	textFontFamily: undefined
});
