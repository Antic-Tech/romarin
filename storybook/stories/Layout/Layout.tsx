import React from 'react';
import { View } from 'react-native';

export const Row = ((props) => {
	const { style, grow, center, ...otherProps, } = props;
	return (
		<View
			style={[
				{
					flex: grow ? 1 : 0,
					flexDirection: 'row',
					justifyContent: center ? 'center': undefined,
				},
				style
			]}
			{...otherProps}
		>
			{props.children}
		</View>
	);
});

export const Column = ((props) => {
	const { style, grow, center, ...otherProps } = props;
	return (
		<View
			style={[
				{
					flex: grow ? 1 : 0,
					flexDirection: 'column',
					alignItems: center ? 'center': undefined,
				},
				style
			]}
			{...otherProps}
		>
			{props.children}
		</View>
	);
});
