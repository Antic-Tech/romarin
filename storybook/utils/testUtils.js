import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
