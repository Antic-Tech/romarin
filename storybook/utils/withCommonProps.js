import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
