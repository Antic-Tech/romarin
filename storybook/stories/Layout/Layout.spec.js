import React from 'react';
import { verifySnapshot } from '../../utils/testUtils';
import { Column, Row } from './Layout';

describe('Row', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Row />);
		});
	});
});

describe('Column', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Column />);
		});
	});
});
