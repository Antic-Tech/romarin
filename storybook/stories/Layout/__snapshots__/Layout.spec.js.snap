// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column Snapshots renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": undefined,
        "flex": 0,
        "flexDirection": "column",
      },
      Object {
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
    ]
  }
/>
`;

exports[`Row Snapshots renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": undefined,
      },
      Object {
        "margin": undefined,
        "marginBottom": undefined,
        "marginHorizontal": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "marginVertical": undefined,
        "padding": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
      },
    ]
  }
/>
`;
